app/src/androidTest/kotlin/at/bitfire/davdroid/servicedetection/CollectionListRefresherTest.kt

```diff
@@ -54,6 +57,12 @@ class CollectionListRefresherTest {
         // The test application is an instance of HiltTestApplication, which doesn't initialize notification channels.
         // However, we need notification channels for the ongoing work notifications.
         NotificationUtils.createChannels(context)
+
+        // Initialize WorkManager for instrumentation tests.
+        val config = Configuration.Builder()
+            .setMinimumLoggingLevel(Log.DEBUG)
+            .build()
+        WorkManagerTestInitHelper.initializeTestWorkManager(context, config)
```

Comment:
Not directly, but as a side effect `PushRegistrationWorker` is launched whenever a collection is changed.

This happens for example when testing/calling `refreshHomesetsAndTheirCollections()` which updates collections via the new repository.

The notification channels are obsolete however, so we don't need to create those.

Reply:
The notification channels are obsolete however, so we don't need to create those.

Reply:
We could remove the `PushRegistrationWorker`'s listener for tests / only install the module for non-tests.

If we only want it for this class:

```
@HiltAndroidTest
@UninstallModules(PushRegistrationWorker.PushRegistrationWorkerModule::class)
class CollectionListRefresherTest {
```

Otherwise we would probably have to [replace the binding with `@TestInstallIn`](https://developer.android.com/training/dependency-injection/hilt-testing#testing-features).

Reply:
Ah yes, nice. I forgot hilt/dagger was capabable of that!